pkg/controllers/termination/controller.go

```diff
@@ -77,24 +77,37 @@ func (c *Controller) Finalize(ctx context.Context, node *v1.Node) (reconcile.Res
 		return reconcile.Result{}, fmt.Errorf("cordoning node, %w", err)
 	}
 	if err := c.terminator.Drain(ctx, node); err != nil {
-		if terminator.IsNodeDrainError(err) {
-			c.recorder.Publish(terminatorevents.NodeFailedToDrain(node, err))
-			return reconcile.Result{Requeue: true}, nil
+		if !terminator.IsNodeDrainError(err) {
+			return reconcile.Result{}, fmt.Errorf("draining node, %w", err)
 		}
-		return reconcile.Result{}, fmt.Errorf("draining node, %w", err)
+		c.recorder.Publish(terminatorevents.NodeFailedToDrain(node, err))
+		// If the underlying machine no longer exists.
+		// If the node exists, requeue after a second to not hammer the API.
+		if _, err := c.cloudProvider.Get(ctx, node.Spec.ProviderID); err != nil {
+			if !cloudprovider.IsMachineNotFoundError(err) {
+				return reconcile.Result{RequeueAfter: 1 * time.Second}, fmt.Errorf("getting machine, %w", err)
```

Comment:
Error takes precedence over all else (see https://github.com/kubernetes-sigs/controller-runtime/blob/a26de2d610c3cf4b2a02688534aaf5a65749c743/pkg/internal/controller/controller.go#L315) so this will be rate-limited regardless of the `RequeueAfter`. I don't think that you need both here anyways